src/monitors/Monitor.v3

```diff
@@ -40,3 +40,23 @@ class Monitor {
 	}
 }
 
+component MonitorUtil {
```

Comment:
Can we hide this as an implementation detail by introducing methods on the `Module` (such as `insertProbeBeforeEntry()` and `insertProbeBeforeReturn()`? That way the interface is uniform through the module and MonitorUtil is an implementation detail.